lib/promscrape/discovery/ec2/instance.go

```diff
@@ -28,8 +35,9 @@ func getReservations(cfg *apiConfig) ([]Reservation, error) {
 	action := "DescribeInstances"
 	var rs []Reservation
 	pageToken := ""
+	apiEndpoint := buildAPIEndpoint(cfg.region, cfg.endpoint, "ec2")
```

Comment:
please move `buildAPIEndpoint()` function call to `getEC2APIResponse()` function, since the returned `apiEndpoint` isn't used outside `getEC2APIResponse`. This may look sub-optimal from performance point of view, since `getEC2APIResponse` is called in the loop. This is OK, since:

* the `buildAPIEndpoint` function is cheap.
* the number of loops is usually small.